### DISCUSSION TITLE
Rugby union Clips into Animation

### DISCUSSION THREAD

**Discussion by alyr0007** [ultralytics/ultralytics#22811] [Q&A]
Hi all,

I’m looking for a faster way to turn rugby video clips into “dots” animations (player-movement diagrams) to help my players see the real footage and the positional map at the same time. I currently do this manually in software like AnimateSlate, but it takes a lot of time.

Does anyone have experience on automating something this?

I'm researching YOLO-based tracker on the video and automatically detect/track all players. It exports the XY coordinate data, which can then be imported straight into AnimateSlate or Metrica to auto-create the animation.

any thoughts really appreciated.

Cheers
Al

**Comment by UltralyticsAssistant** [comment]
👋 Hello @alyr0007, thank you for your interest in Ultralytics and for sharing your rugby analysis idea 🚀 This is an automated response to help you get started quickly — an Ultralytics engineer will also review your discussion and assist you soon 🙂

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, and where many common YOLO workflows (including video processing and tracking-style usage patterns) are already covered.

If this is a 🐛 Bug Report related to YOLO tracking, training, or inference, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can help debug more effectively.

If this is a custom training or workflow ❓ Question (for example, automating player detection, tracking, or exporting coordinates), please include as much detail as possible:
- what model(s) you are using (e.g., YOLO11 variants)
- sample frames or short clips from your rugby videos
- any scripts or command lines you’ve tried
- logs, error messages, or unexpected behaviors you’re seeing  
Also verify you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) if you are training or fine-tuning a model for player detection.

Join the Ultralytics community where it suits you best to discuss ideas like your “dots” animation workflow:
- For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧
- Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/)
- Or explore and share workflows on our [Subreddit](https://www.reddit.com/r/Ultralytics/)  

## Upgrade

Before diving deeper, please upgrade to the latest `ultralytics` package (and make sure all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) are installed) in a **Python>=3.8** environment with **PyTorch>=1.8** to ensure you’re not running into an issue already fixed in a newer release:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by PrashantDixit0** [comment]
> Hi all,
> 
> I’m looking for a faster way to turn rugby video clips into “dots” animations (player-movement diagrams) to help my players see the real footage and the positional map at the same time. I currently do this manually in software like AnimateSlate, but it takes a lot of time.
> 
> Does anyone have experience on automating something this?
> 
> I'm researching YOLO-based tracker on the video and automatically detect/track all players. It exports the XY coordinate data, which can then be imported straight into AnimateSlate or Metrica to auto-create the animation.
> 
> any thoughts really appreciated.
> 
> Cheers Al

@alyr0007 I dont think there is any example solution that present this type of solution, but you can definitely refer to RoboFlow's Sport analytics 
https://github.com/roboflow/sports.

btw @glenn-jocher Should I create a sport analytics PR? If this idea sounds good to you and would be good contribution for Ultralytics audience

**Comment by alyr0007** [comment]
Hi @PrashantDixit0

I just checked out the roboflow link. This is exactly what i am after but for rugby. 

I'm not tech savy i have a limited knowledge (which is dangerous !!) but i know what i'm looking for. 

I'll make contact with them and see what the next steps could be. 

Thanks a million.

Cheers
Al

**Reply by PrashantDixit0** [reply]
If you need help in making it work, I can help you for your Rugby example,
Drop me mail at prasantdixit987@gmail.com ,  if interested

**Reply by glenn-jocher** [reply]
Thanks all, very interesting use-case.  

For this kind of “dots” animation, you can run a YOLO11 tracker over the rugby video, grab per-frame IDs and XY coordinates, and export them (for example as CSV) for AnimateSlate/Metrica to ingest; a minimal pattern in Python would look like:

```python
from ultralytics import YOLO
import csv

model = YOLO("yolo11n.pt")

with open("tracks.csv", "w", newline="") as f:
    writer = csv.writer(f)
    writer.writerow(["frame", "id", "x", "y"])  # center coordinates in pixels

    for frame_idx, result in enumerate(
        model.track(source="path/to/rugby_clip.mp4", stream=True, persist=True)
    ):
        if result.boxes.id is None:
            continue
        xy = result.boxes.xywh[:, :2].cpu().numpy()       # center (x, y)
        ids = result.boxes.id.int().cpu().numpy()         # track IDs
        for (x, y), tid in zip(xy, ids):
            writer.writerow([frame_idx, int(tid), float(x), float(y)])
```

You can then post-process this file into the exact schema your animation tool expects; the tracking APIs and options are described in the [multi-object tracking docs](https://docs.ultralytics.com/modes/track/).  

On the collaboration side: we cannot facilitate or endorse private support via personal email, so please keep help here in Discussions or other public Ultralytics community channels. For the sports analytics idea, a PR with a small, self-contained example (plus brief docs and tests, and without changing `default.yaml`) would be welcome for review.

**Reply by alyr0007** [reply]
Hi @glenn-jocher

Thanks for the prompt reply. 

I'm a rugby coach so all this has gone completely over my head !!! haha.

as a starter do i download Yolo 11 and copy this in? 

sorry i'm outta my depth here !! 

Cheers
Al

**Reply by glenn-jocher** [reply]
Totally okay that this feels overwhelming, Al — at a high level you first need Python and Ultralytics installed, then you can run the little script I shared above.  

As a gentler starting point, try just installing Ultralytics and running tracking from a terminal/command prompt to see it work on a rugby clip:

```bash
pip install -U ultralytics
yolo track model=yolo11n.pt source="path/to/rugby_clip.mp4" save=True
```

This will create an output video with players tracked; once you’re comfortable with that (or have someone a bit more technical beside you), you can move on to adapting the coordinate‑export script. For a step‑by‑step walkthrough of installing and running YOLO11, the [Ultralytics YOLO Quickstart guide](https://docs.ultralytics.com/quickstart/) is a good place to follow along.

**Comment by alyr0007** [comment]
Awesome Glen,

I’ll have a go.

The issue with rugby is that once players get tackled and it’s a ruck it’s
hard to distinguish who’s who that’s the trouble I have had before.  I
think the AI has improved out of sight in the last few years so will to try
this.

Again appreciate your help and insight.

Cheers
Al

**Reply by glenn-jocher** [reply]
You’ve nailed it: rucks and pile‑ups are exactly where any tracker struggles, because players fully occlude each other and IDs will get lost or swapped, so the usual pattern is to get good, clean tracks from open play and accept that ruck segments are noisier; if you later team up with someone more technical, they can look at training/tuning a model on your own footage (e.g., focusing on team colours or jersey numbers) along the lines of the Ultralytics [object tracking overview](https://www.ultralytics.com/glossary/object-tracking), and if you run into questions as you experiment, just drop them here with a short example clip.
